azure_data_cosmos_engine/src/query/mod.rs

```diff
@@ -87,6 +89,144 @@ impl PartitionKeyRange {
+    // TODO: Partition key values here are all currently strings - we need the same sort of PartitionKeyValue
+    // logic used in the main Rust SDK in order to compare and be able to use it within this method.
+    let mut items_by_partition: HashMap<String, Vec<(usize, String, String)>> = HashMap::new();
+    let mut items_by_pk_value: HashMap<String, Vec<(usize, String, String)>> = HashMap::new();
+
+    // Group items by PK value (string or number) - only string for now since we don't have PartitionKeyValue logic yet.
+    for (idx, identity) in item_identities.iter().enumerate() {
+        let pk_value = identity.partition_key_value.clone(); // PartitionKeyValue is enum { String(String), Number(f64) }
+        items_by_pk_value
+            .entry(pk_value.clone())
+            .or_default()
+            .push((idx, identity.id.clone(), pk_value));
+    }
+
+    // For each PK group, compute EPK range and find overlapping ranges
+    for pk_items in items_by_pk_value.values() {
+        let pk_value = &pk_items[0].2;
+        // The Go SDK passes in a pk_value JSON string that comes in as "["value"]". We need to 
+        // get the actual value inside, otherwise hashing fails to get values in the right ranges
+        let inner_value = extract_inner_pk_value(pk_value)?;
+        let pk_value_val = PartitionKeyValue::String(inner_value.clone()); 
+        // TODO: Also needs PK to be updated here
+
+        let epk_range_string =
+            get_hashed_partition_key_string(&[pk_value_val], pk_kind, pk_version);
+            let epk_range = QueryRange {
+                min: epk_range_string.clone(),
+                max: epk_range_string,
+                is_min_inclusive: true,
+                is_max_inclusive: true,
+            };
+        // Here we have to create a clone because get_overlapping_pk_ranges modifies the input pkranges.
+        // For ReadMany we need to keep the full list intact for the next set of items, since a range may be used more than once.
+        let mut pkranges_clone = pkranges.clone();
+        get_overlapping_pk_ranges(&mut pkranges_clone, &[epk_range]);
+        if !pkranges.is_empty() {
+            let range_id = pkranges_clone[0].id.clone();
+            items_by_partition
+                .entry(range_id)
+                .or_default()
+                .extend(pk_items.clone());
+        }
+    }
+    Ok(items_by_partition)
+}
+
+// Creates [`QueryChunk`]s from partitioned items, ensuring no chunk exceeds the maximum item limit.
+// Each chunk is represented as vector of partition key range IDs to lists of [`QueryChunkItem`]s containing the original index, item ID, and partition key value.
+fn create_query_chunks_from_partitioned_items(
+    items_by_partition: &HashMap<String, Vec<(usize, String, String)>>,
+) -> Vec<QueryChunk> {
+    let mut query_chunks: Vec<QueryChunk> = Vec::new();
+    let max_items_per_query = 1000;
```

Comment:
Shouldn't we also be limiting the chunk size if the query string gets to be too long for the service to accept?